paper_test.go

```diff
@@ -0,0 +1,113 @@
+package dropbox
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPaper_List(t *testing.T) {
+	ctx := context.Background()
+	c := client()
+
+	out, err := c.Paper.ListDocs(ctx, &PaperDocsListInput{
+		SortBy:    "modified",
+		SortOrder: "ascending",
+		Limit:     10,
+	})
+
+	require.NoError(t, err)
+	assert.Equal(t, 10, len(out.DocIDs))
+	assert.NotZero(t, out.Cursor.Value)
+	assert.False(t, out.Cursor.Expiration.IsZero())
+}
+
+func TestPaper_Download(t *testing.T) {
+	ctx := context.Background()
+	c := client()
+	var fileID string
+	if fileID = os.Getenv("DROPBOX_PAPER_TOP_LEVEL_FILE_ID"); fileID == "" {
+		t.Skip("No top-level file ID passed in")
+	}
+
+	out, err := c.Paper.Download(ctx, &PaperDownloadInput{
+		DocID:        fileID,
+		ExportFormat: "markdown",
+	})
+	require.NoError(t, err)
+
+	defer out.Body.Close()
+	content, err := ioutil.ReadAll(out.Body)
+	require.NoError(t, err)
+	assert.Equal(t, out.Length, int64(len(content)))
+}
+
+func TestPaper_GetFolderInfoTopLevel(t *testing.T) {
+	ctx := context.Background()
+	c := client()
+
+	var fileID string
+	if fileID = os.Getenv("DROPBOX_PAPER_TOP_LEVEL_FILE_ID"); fileID == "" {
+		t.Skip("No top-level file ID passed in")
+	}
+
+	out, err := c.Paper.GetFolderInfo(ctx, &PaperGetFolderInfoInput{DocID: fileID})
+	require.NoError(t, err)
+	assert.Zero(t, out.FolderSharingPolicyType.Tag)
+	assert.Nil(t, out.Folders)
+}
+
+func TestPaper_GetFolderInfoInsideFolders(t *testing.T) {
+	ctx := context.Background()
+	c := client()
+
+	var fileID string
+	if fileID = os.Getenv("DROPBOX_PAPER_IN_FOLDERS_FILE_ID"); fileID == "" {
+		t.Skip("No top-level file ID passed in")
+	}
+
+	out, err := c.Paper.GetFolderInfo(ctx, &PaperGetFolderInfoInput{DocID: fileID})
+	require.NoError(t, err)
+	assert.Equal(t, "invite_only", out.FolderSharingPolicyType.Tag)
+	require.Len(t, out.Folders, 2)
+	for _, f := range out.Folders {
+		assert.NotZero(t, f.ID)
+		assert.NotZero(t, f.Name)
+	}
+}
+
+/*type PaperGetMetadataOutput struct {
+	DocID           string         `json:"doc_id"`
+	Owner           string         `json:"owner"`
+	Title           string         `json:"title"`
+	CreatedDate     time.Time      `json:"created_date"`
+	Status          PaperDocStatus `json:"status"`
+	Revision        int64          `json:"revision"`
+	LastUpdatedDate time.Time      `json:"last_updated_date"`
+	LastEditor      string         `json:"last_editor"`
+}*/
+
+func TestPaper_AlphaGetMetadata(t *testing.T) {
+	ctx := context.Background()
+	c := client()
+
+	var fileID string
+	if fileID = os.Getenv("DROPBOX_PAPER_TOP_LEVEL_FILE_ID"); fileID == "" {
```

Comment:
How does one get this ID? Can we make it automatic?

Reply:
Those you just pick yourself, just like the token envionment variable used in in `client()`

Reply:
I was tempted to implement the `Upload` endpoint and throw that into the test, but for that you'd still need the identifier for a folder if you want to upload a Paper in a folder, then retrieve it, since there's no create folder method in the Dropbox HTTP docs currently